Add angular interview question (#20)
alibaba-aero/technical-interview: frontend/angular/questions/rxjs.md

# RxJs

- What is Reactive Programming?
- What is RxJS?
- What is Stream in RxJS?

- Observable

  - What is the difference between Cold and Hot Observables?
  - What are the advantages of Observable over Promise?

- Observer

  - What are observers?

- Scheduler

  - What are Schedulers?

- Subscription

  - What is Subscriptions?

- Subject

  - What is Subject?

>   - what is an rxjs subject in Angular?

Reviewer: ```suggestion
  - What is an RxJS subject in Angular?
```

>   - How can we get current value of RxJS Subject or Observable?

Reviewer: ```suggestion
  - How can we get the current value of RxJS Subject or Observable?
```

>   - What are different between of Subject, BehaviorSubject and ReplaySubject?

Reviewer: ```suggestion
  - What are there differences between Subject, BehaviorSubject, and ReplaySubject?
```

- Operators
  - What are RxJS Operators?
  - What is RxJS concatMap?
  - What is RxJS switchMap?
  - What is the use of mergeMap?

>   - When we use zip and combineLatest and withLatestFrom?

Reviewer: ```suggestion
  - When do we use zip and combineLatest and withLatestFrom?
```
